packages/webapp/src/pages/Connection/List.tsx

```diff
@@ -20,59 +20,55 @@ import { useStore } from '../../store';
 
-    useEffect(() => {
-        if (data) {
-            setConnections(data.connections);
-            setFilteredConnections(data.connections);
-            setNumberOfErroredConnections(data.connections.filter((connection) => connection.active_logs).length);
+    const filteredConnections = useMemo<Connection[]>(() => {
+        if (!data) {
+            return [];
         }
-    }, [data]);
-
-    useEffect(() => {
-        if (data) {
-            let filtered = data.connections;
-            if (connectionSearch) {
-                filtered = filtered?.filter((connection) => connection.connection_id.toLowerCase().includes(connectionSearch.toLowerCase()));
-            }
 
-            if (selectedIntegration.length > 0 && !selectedIntegration.includes('all')) {
-                filtered = filtered?.filter((connection) => selectedIntegration.includes(connection.provider_config_key));
-            }
+        const allConnections: Connection[] = data.connections || [];
+        return allConnections.filter((connection) => {
+            const matchesSearch: boolean = !connectionSearch || connection.connection_id.toLowerCase().includes(connectionSearch.toLowerCase());
+            const matchesIntegration: boolean = selectedIntegrations.includes('all') || selectedIntegrations.includes(connection.provider_config_key);
+            const matchesStatus: boolean =
+                selectedStatuses.includes('all') ||
+                (selectedStatuses.includes('ok') && !connection.active_logs) ||
+                (selectedStatuses.includes('error') && !!connection.active_logs);
 
-            if (states.length !== 0 && !states.includes('all') && !(states.includes('ok') && states.includes('error'))) {
-                if (states.includes('error')) {
-                    filtered = filtered?.filter((connection) => connection.active_logs);
-                }
-                if (states.includes('ok')) {
-                    filtered = filtered?.filter((connection) => !connection.active_logs);
-                }
-            }
+            return matchesSearch && matchesIntegration && matchesStatus;
+        });
+    }, [connectionSearch, selectedIntegrations, selectedStatuses, data]);
 
-            setFilteredConnections(filtered || []);
-            setNumberOfErroredConnections((filtered || []).filter((connection) => connection.active_logs).length);
+    const numberOfErroredConnections = useMemo<number>(() => {
+        if (!data) {
+            return 0;
         }
-    }, [connectionSearch, selectedIntegration, states, data]);
 
+        return (data.connections || []).filter((connection) => connection.active_logs).length;
+    }, [data]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
     const debouncedSearch = useCallback(
         debounce((value: string) => {
-            if (!value.trim()) {
-                setConnectionSearch('');
-                setFilteredConnections(data?.connections || []);
```

Comment:
This line was causing a bug where if you searched with an empty/only whitespaced search box, the list of connections displayed would ignore your filters.